--- a/benchmarks/bench_ml.py
+++ b/benchmarks/bench_ml.py
@@ -161,7 +161,7 @@
             samples = np.array(data)
             data = dict()
             data["data"] = samples[:, :-1]
-            data["target"] = np.array(samples[:, -1], dtype=np.int)
+            data["target"] = np.array(samples[:, -1], dtype=np.int64)
     else:
         raise ValueError("dataset not supported.")
     return data["data"], data["target"]
--- a/skopt/space/transformers.py
+++ b/skopt/space/transformers.py
@@ -259,7 +259,7 @@
         if (self.high - self.low) == 0.:
             return X * 0.
         if self.is_int:
-            return (np.round(X).astype(np.int) - self.low) /\
+            return (np.round(X).astype(np.int64) - self.low) /\
                    (self.high - self.low)
         else:
             return (X - self.low) / (self.high - self.low)
@@ -272,5 +272,5 @@
             raise ValueError("All values should be greater than 0.0")
         X_orig = X * (self.high - self.low) + self.low
         if self.is_int:
-            return np.round(X_orig).astype(np.int)
+            return np.round(X_orig).astype(np.int64)
         return X_orig


-- 
